apps/mobile-wallet/src/features/modals/modalSlice.ts

```diff
@@ -18,13 +19,25 @@ const modalSlice = createSlice({
           isClosing: false
         })
       })
-      .addCase(closeModal, (state, { payload: { id } }) => {
-        modalAdapter.updateOne(state, {
-          id,
-          changes: {
-            isClosing: true
+      .addCase(closeModal, (state, { payload: { id, name } }) => {
+        let modalId = id
+
+        if (!modalId && name) {
+          const allModals = selectAll(state)
+          const modalWithName = allModals.find((modal) => modal.params.name === name)
+          if (modalWithName) {
+            modalId = modalWithName.id
           }
-        })
+        }
+
+        if (modalId) {
+          modalAdapter.updateOne(state, {
+            id: modalId,
+            changes: {
+              isClosing: true
+            }
+          })
+        }
```

Comment:
These changes can be removed too if we add the `parentId` modal prop as I proposed above.

Reply:
Done